API/Services/StatisticService.cs

```diff
@@ -539,6 +540,27 @@ public async Task<long> TimeSpentReadingForUsersAsync(IList<int> userIds, IList<
                 p.chapter.AvgHoursToRead * (p.progress.PagesRead / (1.0f * p.chapter.Pages))));
     }
 
+public IEnumerable<StatCount<int>> GetTimeSpentReadingByYear(int userId = 0)
+{
+    var query = _context.AppUserProgresses
+        .Join(_context.Chapter, 
+            progress => progress.ChapterId, 
+            chapter => chapter.Id, 
+            (progress, chapter) => new { Progress = progress, Chapter = chapter })
+        .AsSplitQuery()
+        .AsNoTracking();
+
+    if (userId > 0)
+    {
+        query = query.Where(p => p.Progress.AppUserId == userId);
+    }
+
+    return query.GroupBy(p => p.Progress.LastModified.Year)
+        .OrderBy(g => g.Key)
+        .Select(g => new StatCount<int> { Value = g.Key, Count = g.Sum(x => x.Chapter.AvgHoursToRead) })
```

Comment:
This will need to be a float. In https://github.com/Kareadita/Kavita/pull/3209 I made the AvgHoursToRead a float to fix an issue where anything that was less than 1 hour wasn't counted in stats.